statsmodels/stats/mediation.py

```diff
@@ -0,0 +1,372 @@
+
+            if method == "parametric":
+                # Realization of outcome model parameters from sampling distribution
+                outcome_params = self._simulate_params(outcome_result)
+
+                # Realization of mediation model parameters from sampling distribution
+                mediation_params = self._simulate_params(mediator_result)
+            else:
+                outcome_result = self._fit_model(self.outcome_model,
+                                                 self._outcome_fit_kwargs, boot=True)
+                outcome_params = outcome_result.params
+                mediator_result = self._fit_model(self.mediator_model,
+                                                  self._mediator_fit_kwargs, boot=True)
+                mediation_params = mediator_result.params
+
+            # predicted outcomes[tm][te] is the outcome when the
+            # mediator is set to tm and the outcome/exposure is set to
+            # te.
+            predicted_outcomes = [[None, None], [None, None]]
+            for tm in 0, 1:
+                mex = self._get_mediator_exog(tm)
+                gen = self.mediator_model.get_distribution(mediation_params,
+                                                           mediator_result.scale,
+                                                           exog=mex)
+                potential_mediator = gen.rvs(mex.shape[0])
+
+                for te in 0, 1:
+                    oex = self._get_outcome_exog(te, potential_mediator)
+                    po = self.outcome_model.predict(outcome_params, oex)
+                    predicted_outcomes[tm][te] = po
+
+            for t in 0, 1:
+                indirect_effects[t].append(predicted_outcomes[1][t] - predicted_outcomes[0][t])
+                direct_effects[t].append(predicted_outcomes[t][1] - predicted_outcomes[t][0])
+
+        for t in 0, 1:
+            indirect_effects[t] = np.asarray(indirect_effects[t]).T
+            direct_effects[t] = np.asarray(direct_effects[t]).T
+
+        self.indirect_effects = indirect_effects
+        self.direct_effects = direct_effects
+
+        rslt = MediationResults(self.indirect_effects, self.direct_effects)
+        rslt.method = method
+        return rslt
+
+
+def _pvalue(vec):
+    return 2 * min(sum(vec > 0), sum(vec < 0)) / float(len(vec))
+
+
+class MediationResults(object):
+
+    def __init__(self, indirect_effects, direct_effects):
+
+        self.indirect_effects = indirect_effects
+        self.direct_effects = direct_effects
+
+        indirect_effects_avg = [None, None]
+        direct_effects_avg = [None, None]
+        for t in 0, 1:
+            indirect_effects_avg[t] = indirect_effects[t].mean(0)
+            direct_effects_avg[t] = direct_effects[t].mean(0)
+
+        self.ACME_ctrl = indirect_effects_avg[0]
+        self.ACME_tx = indirect_effects_avg[1]
+        self.ADE_ctrl = direct_effects_avg[0]
+        self.ADE_tx = direct_effects_avg[1]
```

Comment:
since these are user facing attributes, longer more descriptive names would be usefull